src/utils/breakpoint/index.js

```diff
@@ -25,27 +37,48 @@ export function makeLocationId(location: Location) {
 }
 
 export function makePendingLocationId(location: Location) {
+  assertPendingLocation(location);
   const { sourceUrl, line, column } = location;
   const sourceUrlString = sourceUrl || "";
   const columnString = column || "";
+
   return `${sourceUrlString}:${line}:${columnString}`;
 }
 
-export function allBreakpointsDisabled(state) {
-  return state.breakpoints.every(x => x.disabled);
+export function assertBreakpoint(breakpoint: Breakpoint) {
+  assertLocation(breakpoint.location);
+  assertLocation(breakpoint.generatedLocation);
 }
 
-// syncing
-export function equalizeLocationColumn(location, referenceLocation) {
-  if (referenceLocation.column) {
-    return location;
-  }
-  return { ...location, column: undefined };
+export function assertPendingBreakpoint(pendingBreakpoint: PendingBreakpoint) {
+  assertPendingLocation(pendingBreakpoint.location);
+  assertPendingLocation(pendingBreakpoint.generatedLocation);
 }
 
+export function assertLocation(location: Location) {
+  assertPendingLocation(location);
+  const { sourceId } = location;
+  assert(!!sourceId, "location must have a source id");
+}
+
+export function assertPendingLocation(location: PendingLocation) {
+  assert(!!location, "location must exist");
+
+  const { sourceUrl } = location;
+
+  // sourceUrl is null when the source does not have a url
+  assert(sourceUrl !== undefined, "location must have a source url");
+  assert(location.hasOwnProperty("line"), "location must have a line");
+  assert(
+    location.hasOwnProperty("column") != null,
+    "location must have a column"
+  );
+}
```

Comment:
these assertions only run in development and are really helpful for catching the many inconsistencies in data that we had